presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1594,12 +1595,17 @@ protected Scope visitSampledRelation(SampledRelation relation, Optional<Scope> s
             Object samplePercentageObject = samplePercentageEval.optimize(symbol -> {
                 throw new SemanticException(NON_NUMERIC_SAMPLE_PERCENTAGE, relation.getSamplePercentage(), "Sample percentage cannot contain column references");
             });
-
-            if (!(samplePercentageObject instanceof Number)) {
-                throw new SemanticException(NON_NUMERIC_SAMPLE_PERCENTAGE, relation.getSamplePercentage(), "Sample percentage should evaluate to a numeric expression");
+            try {
+                samplePercentageObject = evaluateConstantExpression(relation.getSamplePercentage(), DoubleType.DOUBLE, metadata, session,
```

Comment:
nit: static import DOUBLE. Our general style is to use static imports when possible if it wouldn't create confusion.

Reply:
Done, thx!